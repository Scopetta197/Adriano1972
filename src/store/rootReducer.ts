import { History } from "history";
import { connectRouter } from "connected-react-router";

import userList from "./userList";
import userData from "./userData";

// Use inferred return type for making correctly Redux types
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export default (history: History) => ({
  userList,
  userData,
  router: connectRouter(history) as any,
  // Register more reducers...
});
