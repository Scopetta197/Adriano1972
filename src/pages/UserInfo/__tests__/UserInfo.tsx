import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import { fetchUserDataIfNeed } from "../../../store/userData";
import mockStore from "../../../utils/mockStore";
import UserInfo from "../UserInfo";

describe("<UserInfo />", () => {
  const mockData = {
    id: "1",
    name: "Welly",
    phone: "+886 0970...",
    email: "hivoid19@gmail.com",
    website: "https://wellyshen.com",
  };
  const { id } = mockData;

  const renderHelper = (reducer = {}) => {
    const { dispatch, ProviderWithStore } = mockStore({ userData: reducer });
    const { container } = render(
      <ProviderWithStore>
        <MemoryRouter>
          {/*
            @ts-expect-error */}
          <UserInfo match={{ params: { id } }} />
        </MemoryRouter>
      </ProviderWithStore>
    );

    return { dispatch, firstChild: container.firstChild };
  };

  it("should fetch data when page loaded", () => {
    const { dispatch } = renderHelper();

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(dispatch.mock.calls[0][0].toString()).toBe(
      fetchUserDataIfNeed(id.toString()).toString()
    );
  });

  it("renders the loading status if data invalid", () => {
    expect(renderHelper().firstChild).toMatchSnapshot();
  });

  it("renders the loading status if requesting data", () => {
    const reducer = { [id]: { readyStatus: "request" } };

    expect(renderHelper(reducer).firstChild).toMatchSnapshot();
  });

  it("renders an error if loading failed", () => {
    const reducer = { [id]: { readyStatus: "failure" } };

    expect(renderHelper(reducer).firstChild).toMatchSnapshot();
  });

  it("renders the <Info /> if loading was successful", () => {
    const reducer = { [id]: { readyStatus: "success", item: mockData } };

    expect(renderHelper(reducer).firstChild).toMatchSnapshot();
  });
});
