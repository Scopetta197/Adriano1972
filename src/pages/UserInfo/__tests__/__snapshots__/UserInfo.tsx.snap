// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> renders an error if loading failed 1`] = `
<div
  class="UserInfo"
>
  <p>
    Oops! Failed to load data.
  </p>
</div>
`;

exports[`<UserInfo /> renders the <Info /> if loading was successful 1`] = `
<div
  class="UserInfo"
>
  <div
    class="UserCard"
  >
    <h4>
      User Info
    </h4>
    <ul>
      <li>
        Name: 
        Welly
      </li>
      <li>
        Phone: 
        +886 0970...
      </li>
      <li>
        Email: 
        hivoid19@gmail.com
      </li>
      <li>
        Website: 
        https://wellyshen.com
      </li>
    </ul>
  </div>
</div>
`;

exports[`<UserInfo /> renders the loading status if data invalid 1`] = `
<div
  class="UserInfo"
>
  <p>
    Loading...
  </p>
</div>
`;

exports[`<UserInfo /> renders the loading status if requesting data 1`] = `
<div
  class="UserInfo"
>
  <p>
    Loading...
  </p>
</div>
`;
